addons/sale_timesheet/models/account.py

```diff
@@ -71,9 +71,9 @@ def _check_can_write(self, values):
     def _timesheet_preprocess(self, values):
         values = super(AccountAnalyticLine, self)._timesheet_preprocess(values)
         # task implies so line (at create)
-        if 'task_id' in values and not values.get('so_line') and values.get('employee_id'):
+        if 'task_id' in values and not values.get('so_line') and (values.get('employee_id') or self.employee_id):
             task = self.env['project.task'].sudo().browse(values['task_id'])
-            employee = self.env['hr.employee'].sudo().browse(values['employee_id'])
+            employee = self.employee_id or self.env['hr.employee'].sudo().browse(values['employee_id'])
```

Comment:
I think this is better to keep the priority:
```suggestion
            employee = self.employee_id or self.env['hr.employee'].sudo().browse(values['employee_id']) if values.get('employee_id') else self.employee_id
```